app/Mage.php

```diff
@@ -12,71 +12,95 @@
 
 define('DS', DIRECTORY_SEPARATOR);
 define('PS', PATH_SEPARATOR);
-if (file_exists(__DIR__ . '/vendor/mahocommerce/maho')) {
-    define('BP', dirname(getcwd()));
-} elseif (str_contains(__DIR__, '/vendor/mahocommerce/maho')) {
-    define('BP', str_replace('/vendor/mahocommerce/maho/app', '', __DIR__));
+define('BP', str_replace('/vendor/mahocommerce/maho', '', dirname(__DIR__)));
+
+if (file_exists(BP . '/vendor/mahocommerce/maho')) {
+    define('MAHO_IS_CHILD_PROJECT', true);
+    define('MAHO_FRAMEWORK_DIR', BP . '/vendor/mahocommerce/maho');
 } else {
-    define('BP', dirname(__DIR__));
+    define('MAHO_IS_CHILD_PROJECT', false);
+    define('MAHO_FRAMEWORK_DIR', BP);
 }
 
-Mage::register('original_include_path', get_include_path());
-
 if (!empty($_SERVER['MAGE_IS_DEVELOPER_MODE']) || !empty($_ENV['MAGE_IS_DEVELOPER_MODE'])) {
     Mage::setIsDeveloperMode(true);
     ini_set('display_errors', '1');
     ini_set('error_prepend_string', '<pre>');
     ini_set('error_append_string', '</pre>');
+
+    // Fix for overriding zf1-future during development
+    ini_set('opcache.revalidate_path', 1);
```

Comment:
do we need this if we don't care about the zf1 override? cause in some cases I prefer to enforce hard opcache settings for maximum performance and I wouldn't like to make it impossible to disable revalidate_path

Reply:
I am not sure, it's possible the path cache causes other issues, will have to test. It's basically for any `include` or `require` that uses a relative path, to catch a file that shows up in the path that's different than what was there when the file doing the including was compiled. I don't think it affects Composer's autoloader, but maybe somewhere else in Mage.

In any case, it's absolutely should be enabled in production. I'm not sure how much of an impact it is if you're using developer mode.

Reply:
right! sorry I didn't notice it was only for developer mode! closing this subthread, sorry!

Reply:
No problem, I'll still investigate, because it it does cause an issue, then users should be aware that they may need to flush the opcache when they re-deploy. But that wouldn't be needed on every deploy, just in some cases.

Reply:
same thing happens whenever you tweak opcache (which you should do), if you have a revalidate_freq to 0 (which in production is amazing) you have to reset opcache manually